doc/specs/#2557 - Settings Keybinding.md

```diff
@@ -0,0 +1,120 @@
+---
+author: Carlos Zamora @carlos-zamora
+created on: 2020-05-14
+last updated: 2020-05-14
+issue id: #2557
+---
+
+# Open Settings Keybinding
+
+## Abstract
+
+This spec outlines an expansion to the existing `openSettings` keybinding.
+
+## Inspiration
+
+As a Settings UI becomes more of a reality, the behavior of this keybinding will be expanded on to better interact with the UI. Prior to a Settings UI, there was only one concept of the modifiable user settings: settings.json.
+
+Once the Settings UI is created, we can expect users to want to access the following scenarios:
+- Settings UI: globals page
+- Settings UI: profiles page
+- Settings UI: color schemes page
+- Settings UI: keybindings page
+- settings.json
+- defaults.json
+
+
+## Solution Design
+Originally, #2557 was intended to allow for a keybinding arg to access defaults.json. I imagined a keybinding arg such as "openDefaults: true/false" to accomplish this. However, this is not expandable in the following scenarios:
+- what if we decide to create more settings files in the future? (i.e. themes.json, extensions.json, etc...)
+- when the Settings UI comes in, there is ambiguity as to what `openSettings` does (json? UI? Which page?)
+
+### Proposition 1: the `target` arg
+Instead, what if we introduced a new `target` keybinding argument, that could be used as follows:
+| Keybinding Command | Behavior |
+|--|--|
+| `"command": { "action": "openSettings", "target": "jsonSettings" }`   | opens "settings.json" in your default text editor |
+| `"command": { "action": "openSettings", "target": "jsonDefaults" }`   | opens "defaults.json" in your default text editor |

```

Comment:
similarly, `defaultsFile`? I know that there's no chance we're moving off json, but still 🤔